### ISSUE TITLE
PRCC上clothes changed的评估

### ISSUE THREAD

**Issue by YuSuen** [fountaindream/DLAW#2]
我将这里的https://github.com/fountaindream/DLAW/blob/b4bbf26edf68350f0f11b13fc6d4f8fc2a1d1795/data_prcc.py#L66C68-L66C68 new_B改为new_C来进行clothes changed场景的评估，其结果与论文报告值差异较大，请问我是否遗漏了哪里还需要修改的地方，希望可以帮忙指出

这是CC场景测试结果
![捕获-cc](https://github.com/fountaindream/DLAW/assets/33176642/70a69047-95a1-45c9-bcb5-24dce8a11798)
这是SC场景测试结果
![捕获-sc](https://github.com/fountaindream/DLAW/assets/33176642/a0eb774a-6256-4baf-9655-23a8dc6376f6)

**Comment by fountaindream** [comment]
Try using two GPUs for evaluation instead.

**Comment by YuSuen** [comment]
作者你好：


非常感谢你的回复！


我确实是使用单GPU测试的，因为在使用两个GPU时会遇到下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！







Yu-Suen
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fountaindream/DLAW"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月20日(星期五) 晚上6:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fountaindream/DLAW] PRCC上clothes changed的评估 (Issue #2)





 
Try using two GPUs for evaluation instead.

**Comment by YuSuen** [comment]
非常感谢你的回复！
我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！
![issue](https://github.com/fountaindream/DLAW/assets/33176642/3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1)

**Comment by fountaindream** [comment]
> 非常感谢你的回复！ 我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！ ![issue](https://user-images.githubusercontent.com/33176642/276914548-3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1.PNG)

You may change the batch size for testing in prcc.opt . For New_B 16 and for New_C 14.

**Comment by YuSuen** [comment]
> > 非常感谢你的回复！ 我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！ ![issue](https://user-images.githubusercontent.com/33176642/276914548-3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1.PNG)
> 
> You may change the batch size for testing in prcc.opt . For New_B 16 and for New_C 14.

我已经重新测试了，但是和单GPU的结果差异不大

![捕获-14](https://github.com/fountaindream/DLAW/assets/33176642/90212310-0534-4e74-b776-37707055dd07)

**Comment by fountaindream** [comment]
> > > 非常感谢你的回复！ 我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！ ![issue](https://user-images.githubusercontent.com/33176642/276914548-3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1.PNG)
> > 
> > 
> > You may change the batch size for testing in prcc.opt . For New_B 16 and for New_C 14.
> 
> 我已经重新测试了，但是和单GPU的结果差异不大
> 
> ![捕获-14](https://user-images.githubusercontent.com/33176642/276920001-90212310-0534-4e74-b776-37707055dd07.PNG)

I retrain the codes and obtain the results as below， and it seems that all work well.
![image](https://github.com/fountaindream/DLAW/assets/31584430/a50ad7b7-2c67-4a68-b140-94514c44459e)

**Comment by YuSuen** [comment]
> > > > 非常感谢你的回复！ 我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！ ![issue](https://user-images.githubusercontent.com/33176642/276914548-3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1.PNG)
> > > 
> > > 
> > > You may change the batch size for testing in prcc.opt . For New_B 16 and for New_C 14.
> > 
> > 
> > 我已经重新测试了，但是和单GPU的结果差异不大
> > ![捕获-14](https://user-images.githubusercontent.com/33176642/276920001-90212310-0534-4e74-b776-37707055dd07.PNG)
> 
> I retrain the codes and obtain the results as below， and it seems that all work well. ![image](https://user-images.githubusercontent.com/31584430/276931919-a50ad7b7-2c67-4a68-b140-94514c44459e.png)

那就很奇怪了，我使用的是3090gpu，但在ltcc上获得的结果差异不是很大；
请问在跑prcc的代码时哪里还有要特别注意的吗？

**Comment by fountaindream** [comment]
> > > > > 非常感谢你的回复！ 我确实是使用单GPU测试的，因为在使用两个GPU时我遇到了下面问题，这貌似和多GPU数据分配相关，我没有找到很好的办法解决，请问你是否遇到过这个问题，能否帮忙解决，十分感谢！ ![issue](https://user-images.githubusercontent.com/33176642/276914548-3e1aabd8-2be6-489e-b17d-ee1d4cdff8e1.PNG)
> > > > 
> > > > 
> > > > You may change the batch size for testing in prcc.opt . For New_B 16 and for New_C 14.
> > > 
> > > 
> > > 我已经重新测试了，但是和单GPU的结果差异不大
> > > ![捕获-14](https://user-images.githubusercontent.com/33176642/276920001-90212310-0534-4e74-b776-37707055dd07.PNG)
> > 
> > 
> > I retrain the codes and obtain the results as below， and it seems that all work well. ![image](https://user-images.githubusercontent.com/31584430/276931919-a50ad7b7-2c67-4a68-b140-94514c44459e.png)
> 
> 那就很奇怪了，我使用的是3090gpu，但在ltcc上获得的结果差异不是很大； 请问在跑prcc的代码时哪里还有要特别注意的吗？

1) You may change the margin for the weighted triplet loss (triplet loss1) as 1.2.
2) You may decrease the weight for the similar suppression since the outfits for each person in prcc are too few.

**Comment by xz-zhou** [comment]
@fountaindream Hi, I also haven't achieved over 50% in Rank1 and mAP results on the PRCC dataset while using a single RTX 4080. Can you provide a pre-trained model?

**Comment by xz-zhou** [comment]
Thank you very much, I've seen your update.

**Comment by ForeUP** [comment]
@YuSuen @xz-zhou 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 `Rank-1` 都和论文差距很大，`mAP` 基本一致，难道是我的数据集有问题？
这是我测试出来的结果：
> Cloth-change: Rank@1:0.438134 Rank@5:0.464503 Rank@10:0.531440 mAP:0.351161;
Standard: Rank@1:0.551724 Rank@5:0.592292 Rank@10:0.649087 mAP:0.404388

**Comment by gf2020whj** [comment]
> 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 都和论文差距很大， 基本一致，难道是我的数据集有问题？这是我测试出来的结果：`Rank-1``mAP`
> 
> > 换布：Rank@1：0.438134 Rank@5：0.464503 Rank@10：0.531440 mAP：0.351161;标准： Rank@1：0.551724 Rank@5：0.592292 Rank@10：0.649087 mAP：0.404388

我也是这个情况 ，请问你解决了吗

**Comment by YuSuen** [comment]
> > 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 都和论文差距很大， 基本一致，难道是我的数据集有问题？这是我测试出来的结果：`` Rank-1``mAP ``
> > > 换布：Rank@1：0.438134 Rank@5：0.464503 Rank@10：0.531440 mAP：0.351161;标准： Rank@1：0.551724 Rank@5：0.592292 Rank@10：0.649087 mAP：0.404388
> 
> 我也是这个情况 ，请问你解决了吗

没有，换个模型吧

**Comment by gf2020whj** [comment]
> > > 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 都和论文差距很大， 基本一致，难道是我的数据集有问题？这是我测试出来的结果：mAP ''` Rank-1`
> > > > 换布：Rank@1：0.438134 Rank@5：0.464503 Rank@10：0.531440 mAP：0.351161;标准： Rank@1：0.551724 Rank@5：0.592292 Rank@10：0.649087 mAP：0.404388
> > 
> > 
> > 我也是这个情况 ，请问你解决了吗
> 
> 没有，换个模型吧
请问作者给出的prcc数据集，new_A,new_B,new_C,new_train,new_val分别是在原始数据集上做了什么操作，我看官方给出的prcc数据集是train,test,val，期待您的帮助，谢谢。

**Comment by YuSuen** [comment]
> > > > 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 都和论文差距很大， 基本一致，难道是我的数据集有问题？这是我测试出来的结果：mAP ''` Rank-1`
> > > > > 换布：Rank@1：0.438134 Rank@5：0.464503 Rank@10：0.531440 mAP：0.351161;标准： Rank@1：0.551724 Rank@5：0.592292 Rank@10：0.649087 mAP：0.404388
> > > 
> > > 
> > > 我也是这个情况 ，请问你解决了吗
> > 
> > 
> > 没有，换个模型吧
> > 请问作者给出的prcc数据集，new_A,new_B,new_C,new_train,new_val分别是在原始数据集上做了什么操作，我看官方给出的prcc数据集是train,test,val，期待您的帮助，谢谢。

作者在readme给了他们处理过的数据，有问题直接联系作者吧，我已经不做这个了

**Comment by YuSuen** [comment]
> > > > 你们LTCC数据集复现的结果是多少啊？我用作者提供的预训练模型和我自己跑了500轮的结果的 都和论文差距很大， 基本一致，难道是我的数据集有问题？这是我测试出来的结果：mAP ''` Rank-1`
> > > > > 换布：Rank@1：0.438134 Rank@5：0.464503 Rank@10：0.531440 mAP：0.351161;标准： Rank@1：0.551724 Rank@5：0.592292 Rank@10：0.649087 mAP：0.404388
> > > 
> > > 
> > > 我也是这个情况 ，请问你解决了吗
> > 
> > 
> > 没有，换个模型吧
> > 请问作者给出的prcc数据集，new_A,new_B,new_C,new_train,new_val分别是在原始数据集上做了什么操作，我看官方给出的prcc数据集是train,test,val，期待您的帮助，谢谢。

做了parsing，有问题直接联系作者吧，我已经不做这个了
